content-tools/update-cdids/main.go

```diff
@@ -0,0 +1,443 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"github.com/ONSdigital/log.go/log"
+	"github.com/tealeg/xlsx"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
+)
+
+const (
+	ZebedeeToken = "ZEBEDEE_TOKEN"
+	FlorenceToken = "X-Florence-Token"
+)
+
+var (
+	zebedeeURL  string
+	environment string
+	username    string
+	password    string
+	filePath    string
+	sheetname   string
+	limit       int64
+)
+
+type CollectionDescription struct {
+	Name            string      `json:"name"`
+	Type            string      `json:"type"`
+	PublishDate     interface{} `json:"publishDate"`
+	Teams           []string    `json:"teams"`
+	CollectionOwner string      `json:"collectionOwner"`
+	ReleaseURI      interface{} `json:"releaseUri"`
+	IsEncrypted     bool        `json:"isEncrypted"`
+}
+
+type Credential struct {
+	Email    string `json:"email"`
+	Password string `json:"password"`
+}
+
+type CollectionDescriptionResponse struct {
+	ID string `json:"id"`
+}
+
+func main() {
+	setupFlags()
+
+	ctx := context.Background()
+	httpClient := http.DefaultClient
+
+	if validateMandatoryParams(ctx) {
+		return
+	}
+
+	cdIDPairs := readCdIDPairs(ctx, filePath, sheetname, limit)
+	for _, pair := range cdIDPairs {
+		pair.Print(ctx, "Parsed ")
+	}
+	log.Event(ctx, fmt.Sprintf("Read %d cdIDPairs" , len(cdIDPairs)), log.INFO)
+	ctx, err := authenticate(ctx, httpClient, environment, username, password)
+	if err != nil {
+		_ = fmt.Errorf("error occurred while authenticating. Stopping the script. error: %s", err.Error())
+		return
+	}
+
+	collectionID, err := createCollection(ctx, httpClient, getCollectionName(), environment)
+	if err != nil {
+		fmt.Errorf("error occurred while creating collection. Stopping the script. error: %s", err.Error())
```

Comment:
Same here `fmt.ErrorF` creates an error with the formatted string as its message - you aren't assigning this return value to anything so its never used.